src/main/java/org/junit/internal/runners/ErrorReportingRunner.java

```diff
@@ -13,19 +13,28 @@
 public class ErrorReportingRunner extends Runner {
     private final List<Throwable> causes;
 
-    private final Class<?> testClass;
+    private final String classNames;
 
     public ErrorReportingRunner(Class<?> testClass, Throwable cause) {
-        if (testClass == null) {
-            throw new NullPointerException("Test class cannot be null");
+        this(cause, new Class<?>[] { testClass });
+    }
+    
+    public ErrorReportingRunner(Throwable cause, Class<?>... testClasses) {
+        if (testClasses == null || testClasses.length == 0) {
+            throw new NullPointerException("Test classes cannot be null or empty");
+        }
+        for (Class<?> testClass : testClasses) {
+            if (testClass == null) {
+                throw new NullPointerException("Test class cannot be null");
+            }
         }
-        this.testClass = testClass;
+        classNames = getClassNames(testClasses);
         causes = getCauses(cause);
     }
-
+    
     @Override
     public Description getDescription() {
-        Description description = Description.createSuiteDescription(testClass);
+        Description description = Description.createSuiteDescription(classNames);
```

Comment:
This is an externally visible change for the case where the existing constructor is used.

Instead of making suggestions here, I'm going to create a pull with your commits and one additional commit to fix this and to fix some other minor issues.

The other pull is just to make reviewing my proposed changes easier; I still plan on merging your pull. Details to follow soon.